docs/source/upgrade.rst

```diff
@@ -50,6 +50,18 @@ This kind of error usually occurs by mistake (for example, not performing ``to_g
    As the argument validation is strictened, call of functions intentionally mixing NumPy/CuPy arrays in arguments will not work in Chainer v4.
    Please transfer all arrays to the same device before calling functions.
 
+Reference to Function Nodes Not Retained in Function Hooks
```

Comment:
* "Reference" -> "References"
* This gives the impression that all function hooks are affected by this change. How about "...Not Retained in TimerHook and CupyMemoryProfilerHook"?